hyperspy/tests/axes/test_axes_manager.py

```diff
@@ -75,3 +77,102 @@ def test_update_from(self):
+        signal_axis = s.axes_manager.signal_axes[0]
+        signal_extent = (signal_axis.low_value, signal_axis.high_value)
+        assert signal_extent == s.axes_manager.signal_extent
+        assert len(s.axes_manager.navigation_extent) == 0
+        assert () == s.axes_manager.navigation_extent
+
+    def test_2d_signal1d(self):
+        s = Signal1D(arange(100).reshape(10, 10))
+        assert len(s.axes_manager.signal_extent) == 2
+        signal_axis = s.axes_manager.signal_axes[0]
+        signal_extent = (signal_axis.low_value, signal_axis.high_value)
+        assert signal_extent == s.axes_manager.signal_extent
+        assert len(s.axes_manager.navigation_extent) == 2
+        nav_axis = s.axes_manager.navigation_axes[0]
+        nav_extent = (nav_axis.low_value, nav_axis.high_value)
+        assert nav_extent == s.axes_manager.navigation_extent
+
+    def test_3d_signal1d(self):
+        s = Signal1D(arange(1000).reshape(10, 10, 10))
+        assert len(s.axes_manager.signal_extent) == 2
+        signal_axis = s.axes_manager.signal_axes[0]
+        signal_extent = (signal_axis.low_value, signal_axis.high_value)
+        assert signal_extent == s.axes_manager.signal_extent
+        assert len(s.axes_manager.navigation_extent) == 4
+        nav_axis0 = s.axes_manager.navigation_axes[0]
+        nav_axis1 = s.axes_manager.navigation_axes[1]
+        nav_extent = (
+                nav_axis0.low_value, nav_axis0.high_value,
+                nav_axis1.low_value, nav_axis1.high_value,
+                )
+        assert nav_extent == s.axes_manager.navigation_extent
+
+    def test_2d_signal2d(self):
+        s = Signal2D(arange(100).reshape(10, 10))
+        assert len(s.axes_manager.signal_extent) == 4
+        signal_axis0 = s.axes_manager.signal_axes[0]
+        signal_axis1 = s.axes_manager.signal_axes[1]
+        signal_extent = (
+                signal_axis0.low_value, signal_axis0.high_value,
+                signal_axis1.low_value, signal_axis1.high_value,
+                )
+        assert signal_extent == s.axes_manager.signal_extent
+        assert len(s.axes_manager.navigation_extent) == 0
+        assert () == s.axes_manager.navigation_extent
+
+    def test_3d_signal2d(self):
+        s = Signal2D(arange(1000).reshape(10, 10, 10))
+        assert len(s.axes_manager.signal_extent) == 4
+        signal_axis0 = s.axes_manager.signal_axes[0]
+        signal_axis1 = s.axes_manager.signal_axes[1]
+        signal_extent = (
+                signal_axis0.low_value, signal_axis0.high_value,
+                signal_axis1.low_value, signal_axis1.high_value,
+                )
+        assert signal_extent == s.axes_manager.signal_extent
+        assert len(s.axes_manager.navigation_extent) == 2
+        nav_axis = s.axes_manager.navigation_axes[0]
+        nav_extent = (nav_axis.low_value, nav_axis.high_value)
+        assert nav_extent == s.axes_manager.navigation_extent
+
+    def test_changing_scale_offset(self):
```

Comment:
I added some tests for checking this: https://github.com/hyperspy/hyperspy/pull/1508/files#diff-c962fa54b5ae9b1da4d5667347ccd04cR82